hathor/builder/builder.py

```diff
@@ -239,30 +245,82 @@ def _get_or_create_pubsub(self) -> PubSubManager:
 
     def _get_or_create_rocksdb_storage(self) -> RocksDBStorage:
         assert self._rocksdb_path is not None
-        if self._rocksdb_storage is None:
-            self._rocksdb_storage = RocksDBStorage(path=self._rocksdb_path)
+
+        if self._rocksdb_storage is not None:
+            return self._rocksdb_storage
+
+        kwargs = {}
+        if self._rocksdb_cache_capacity is not None:
+            kwargs = dict(cache_capacity=self._rocksdb_cache_capacity)
+
+        self._rocksdb_storage = RocksDBStorage(
+            path=self._rocksdb_path,
+            **kwargs
+        )
+
         return self._rocksdb_storage
 
     def _get_or_create_tx_storage(self) -> TransactionStorage:
         if self._tx_storage is not None:
             return self._tx_storage
+
         if self._storage_type == StorageType.MEMORY:
             return TransactionMemoryStorage()
-        elif self._storage_type == StorageType.ROCKSDB:
+
+        if self._storage_type == StorageType.ROCKSDB:
             rocksdb_storage = self._get_or_create_rocksdb_storage()
             use_memory_index = self._force_memory_index
-            return TransactionRocksDBStorage(rocksdb_storage, use_memory_indexes=use_memory_index)
+
+            kwargs = {}
+            if self._with_rocksdb_index is not None:
+                kwargs = dict(with_index=self._with_rocksdb_index)
+
+            return TransactionRocksDBStorage(
+                rocksdb_storage,
+                use_memory_indexes=use_memory_index,
+                **kwargs
+            )
+
         raise NotImplementedError
 
+    def _get_event_storage(self) -> EventStorage:
+        if self._storage_type == StorageType.MEMORY:
+            return EventMemoryStorage()
+
+        if self._storage_type == StorageType.ROCKSDB:
+            rocksdb_storage = self._get_or_create_rocksdb_storage()
+            return EventRocksDBStorage(rocksdb_storage)
+
+        raise NotImplementedError
+
+    def with_event_manager(self, *, event_ws_factory: EventWebsocketFactory) -> 'Builder':
+        self.check_if_can_modify()
+
+        self._event_manager = EventManager(
+            reactor=self._get_reactor(),
+            pubsub=self._get_or_create_pubsub(),
+            event_storage=self._get_event_storage(),
+            event_ws_factory=event_ws_factory
+        )
+
+        return self
+
     def use_memory(self) -> 'Builder':
         self.check_if_can_modify()
         self._storage_type = StorageType.MEMORY
         return self
 
-    def use_rocksdb(self, path: str) -> 'Builder':
+    def use_rocksdb(
+        self,
+        path: str,
+        with_index: Optional[bool] = None,
+        cache_capacity: Optional[int] = None
+    ) -> 'Builder':
         self.check_if_can_modify()
         self._storage_type = StorageType.ROCKSDB
         self._rocksdb_path = path
+        self._with_rocksdb_index = with_index
```

Comment:
Isn't be better to standardize with the `_rocksdb_` prefix? For instance, this one would be `_rocksdb_with_index`.

Reply:
Agreed. Done in https://github.com/HathorNetwork/hathor-core/pull/568/commits/33a1b40d435bb62afe13891b4c418ab5667bb54e

Reply:
Thanks!